src/python_testing/TC_ACL_2_3.py

```diff
@@ -0,0 +1,331 @@
+                     "Result is SUCCESS"),
+            TestStep(7, "TH1 reads DUT Endpoint 0 AccessControl cluster Extension attribute",
+                     "Result is SUCCESS, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_OK_SINGLE, fabric index is F1"),
+            TestStep(8, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_OK_FULL",
+                     "Result is SUCCESS"),
+            TestStep(9, "TH1 reads DUT Endpoint 0 AccessControl cluster Extension attribute",
+                     "Result is SUCCESS, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_OK_FULL, fabric index is F1"),
+            TestStep(10, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_LENGTH",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(11, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_STRUCT",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(12, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_LIST",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(13, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_ELEM",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(14, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_OVERFLOW",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(15, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_UNDERFLOW",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(16, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 1 element struct data field: D_BAD_NONE",
+                     "Result is CONSTRAINT_ERROR (0x87)"),
+            TestStep(17, "TH1 writes DUT Endpoint 0 AccessControl cluster Extension attribute, value is list of AccessControlExtensionStruct containing 2 elements struct data field: D_OK_EMPTY, D_OK_SINGLE",
+                     "Result is SUCCESS for first struct, value is D_OK_EMPTY, fabric index is F1, CONTSTRAINT_ERROR for second struct"),
```

Comment:
Sorry I forgot to change this in the Test-plan, will change it now:
```suggestion
                     "Result is CONSTRAINT_ERROR (0x87) for the entire list"),
```